src/EFCore.PG.NTS/Query/ExpressionTranslators/Internal/NpgsqlNetTopologySuiteMethodCallTranslatorPlugin.cs

```diff
@@ -28,94 +31,91 @@ public class NpgsqlNetTopologySuiteMethodCallTranslatorPlugin : IMethodCallTrans
     public class NpgsqlGeometryMethodTranslator : IMethodCallTranslator
     {
         static readonly MethodInfo _collectionItem =
-            typeof(IGeometryCollection).GetRuntimeProperty("Item").GetMethod;
+            typeof(GeometryCollection).GetRuntimeProperty("Item").GetMethod;
+
+        readonly ISqlExpressionFactory _sqlExpressionFactory;
+        readonly IRelationalTypeMappingSource _typeMappingSource;
+
+        public NpgsqlGeometryMethodTranslator(ISqlExpressionFactory sqlExpressionFactory, IRelationalTypeMappingSource typeMappingSource)
+        {
+            _sqlExpressionFactory = sqlExpressionFactory;
+            _typeMappingSource = typeMappingSource;
+        }
 
         /// <inheritdoc />
         [CanBeNull]
-        public virtual Expression Translate(MethodCallExpression e, IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+        public virtual SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadOnlyList<SqlExpression> arguments)
         {
-            if (!typeof(IGeometry).IsAssignableFrom(e.Method.DeclaringType))
+            if (!typeof(Geometry).IsAssignableFrom(method.DeclaringType))
                 return null;
```

Comment:
This condition repeated many times, so probably there should be a base type for geometry translators.

Reply:
You mean the other time in the member translator? Not sure that merits a base class... Let's leave as-is.